mailing/sendgrid.go

```diff
@@ -0,0 +1,45 @@
+package mailing
+
+import (
+	"context"
+	"fmt"
+	"github.com/gazebo-web/gz-go/v8"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+	"net/http"
+)
+
+// sendgridEmailService implements the Sender interface using Sendgrid.
+//
+//	Reference: https://github.com/sendgrid/sendgrid-go
+type sendgridEmailService struct {
+	client sendgridSender
+}
+
+// Send sends an email from sender to the given recipients. The email body is described by a template
+// that contains the information passed through data.
+func (s *sendgridEmailService) Send(ctx context.Context, recipients []string, sender, subject, template string, data any) error {
+	from := mail.NewEmail("", sender)
+	htmlContent, err := gz.ParseHTMLTemplate(template, data)
+	if err != nil {
+		return err
+	}
+	for _, recipient := range recipients {
+		to := mail.NewEmail("", recipient)
+		m := mail.NewSingleEmail(from, subject, to, "", htmlContent)
+		res, err := s.client.SendWithContext(ctx, m)
+		if err != nil {
+			return err
+		}
+		if res.StatusCode != http.StatusOK {
+			return fmt.Errorf("failed to send email: (%d) %s", res.StatusCode, http.StatusText(res.StatusCode))
+		}
+	}
```

Comment:
Related to my interface comment, this isn't immediately necessary, but this implementation does not allow sending an email to more than one recipient at a time. It allows sending the same email to multiple recipients, but not a single email with multiple recipients that can respond/see each other.

We should use the `recipients` and optionally `ccs` and `bccs` arguments to configure recipients.

In addition, `NewSingleEmail` wraps `NewV3MailInit`, but is a bit limiting in the options it provides, and even then we don't use all of its options. 

Since we're going to always send an HTML template, and in order to support multiple recipients/CCs/BCCs, we can change this to

```suggestion
	p := mail.NewPersonalization()
	p.AddFrom(mail.NewEmail("", address))
	p.AddTos(stringsToEmails(recipients))
	p.AddCCs(stringsToEmails(ccs))
	p.AddBCCs(stringsToEmails(bccs))
	p.Subject = subject
	
	m := mail.NewV3Mail()
	m.AddPersonalizations(p)
	m.AddContent(mail.NewContent("text/html", htmlContent))
	
	res, err := s.client.SendWithContext(ctx, m)
	if err != nil {
		return err
	}
	if res.StatusCode != http.StatusOK {
		return fmt.Errorf("failed to send email: (%d) %s", res.StatusCode, http.StatusText(res.StatusCode))
	}
```

Where `stringsToEmails` is

```go
// stringsToEmails converts a set of emails represented by a slice of strings to a slice of *mail.Email.
func stringsToEmails(strings []string) []*mail.Email {
	emails := make([]*mail.Email, len(strings)
	for i, email := range strings {
		emails[i] = mail.NewEmail("", email)
	}
	return emails
}
```

Notice I have removed the `for` as it is no longer necessary. If someone wants to send an email to multiple, single recipients, they are expected to add a `for` in their application-specific code.

Reply:
I've implemented everything on this comment except for the `cc` and `bcc` support, as we're not planning on adding that right now.

Reply:
I guess I just feel it's a minimal amount of work. It's basically passing parameters. Tests should only check that the addresses are there, and should be similar to copying and pasting tests for `recipients`.

I don't mind going as is, but again, the amount of work required is minimal.